service/storage_service.cc

```diff
@@ -871,7 +871,7 @@ class topology_coordinator {
                 builder.set_transition_state(topology::transition_state::write_both_read_old)
                        .set_current_cdc_generation_id(cdc_gen_id);
                 auto str = fmt::format("{}: committed new CDC generation, ID: {}", node.rs->state, cdc_gen_id);
-                co_await update_replica_state(std::move(node), {builder.build()}, std::move(str));
+                co_await update_topology_state(std::move(node.guard), {builder.build()}, std::move(str));
```

Comment:
I wanted to have this guaranty that lifetime of `node_to_work_on` will be the same as the guard inside it (until the most internal user moves guard into the raft command and discard the node), but hear, buy moving `node.guard` only the guaranty is broken :(

Reply:
We do `break` and leave the function immediately after this line, so `node_to_work_on` is also destroyed, but I get what you mean.

WDYT about introducing `update_replica_state(node_to_work_on&&, ...)` which will just call `update_topology_state` inside and pass the guard?

I'll need the more general `update_topology_state` for the CDC PR - where I modify this case so it doesn't use `node_to_work_on`.

Reply:
We can add a function that gets `node_to_work_on&&` and returns the `guard`.